morgoth/cli.py

```diff
@@ -147,54 +147,56 @@ def make_release(xpi_file, bearer, profile, verbose, reupload):
             output('Release could not be auto-generated.', Fore.RED)
             exit(1)
 
-        session = boto3.Session(profile_name=profile)
-        s3 = session.resource('s3')
-        bucket = s3.Bucket(settings.get('aws.bucket_name', DEFAULT_AWS_BUCKET_NAME))
-
-        existing_files = []
-        for obj in bucket.objects.filter(Prefix=prefix):
-            existing_files.append(obj.key)
-
-        suffix = ''
-        upload_path = xpi.get_ftp_path(prefix)
-        exists = upload_path in existing_files
-
-        uploaded = False
-        if exists:
-            if validate_uploaded_xpi_hash(xpi, bucket, upload_path):
-                uploaded = True
-            else:
-                index = 2
-                check_path = xpi.get_ftp_path(prefix, suffix='-{}'.format(index))
-                while check_path in existing_files:
-                    if validate_uploaded_xpi_hash(xpi, bucket, check_path):
-                        uploaded = True
-                        suffix = '-{}'.format(index)
-                        break
-                    index += 1
-                    check_path = xpi.get_ftp_path(prefix, suffix='-{}'.format(index))
+        if not xpi.archived:
```

Comment:
The diff looks convoluted... but the logic remains the same. I just enclosed the logic that uploads the file as is within the `if not xpi.archived` block